src/coldcard.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Provides classes for interacting with a Coldcard via TXT/JSON/PSBT files
+ *
+ * The following API classes are implemented:
+ *
+ * * ColdcardExportExtendedPublicKey
+ * * ColdcardSignMultisigTransaction
+ * * ColdcardMultisigWalletConfig
+ *
+ * @module coldcard
+ */
+import {
+  IndirectKeystoreInteraction,
+} from "./interaction";
+
+export const COLDCARD = 'coldcard';
+export const WALLET_CONFIG_VERSION = "0.0.1";
+
+
+/**
+ * Base class for indirect interactions with Coldcard.
+ *
+ * @extends {module:interaction.IndirectKeystoreInteraction}
+ */
+export class ColdcardInteraction extends IndirectKeystoreInteraction {
+
+}
+
+/**
+ * Base class for coldcard interactions (reading/writing files)
+ *
+ * @extends {module:hermit.HermitInteraction}
+ */
+export class ColdcardFileReader extends ColdcardInteraction {
+
+}
+
+export class ColdcardExportExtendedPublicKey extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns signature request data to display via PSBT for Coldcard and returns the PSBT
+ * data back to the calling application
+ */
+export class ColdcardSignTransaction extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns a valid multisig wallet config text file to send over to a Coldcard
+ *
+ * NOTE: only the root xfp of the signing device is required to be correct, but we
+ * recommend only setting up the multisig wallet on the Coldcard with complete xfp
+ * information. We define isSupported() to check exactly this condition.
+ *
+ * This is an example Coldcard config file from
+ * https://coldcardwallet.com/docs/multisig
+ *
+ * # Coldcard Multisig setup file (exported from 4369050F)
+ * #
+ * Name: MeMyself
+ * Policy: 2 of 4
+ * Derivation: m/45'
+ * Format: P2WSH
+ *
+ * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+ * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+ * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+ * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+ *
+ */
+export class ColdcardMultisigWalletConfig {
+  constructor({jsonConfig}) {
+    if (jsonConfig &&
+      jsonConfig.extendedPublicKeys &&
```

Comment:
Why leave these in the `jsonConfig`? to me it feels like once initialized from the json that's passed in, these become properties of the config itself. `configInstance.jsonconfig.extendedPublicKeys` feels a little unwieldy if you're trying to access. This becomes even nicer for internal access, e.g. `this.extendedPublicKeys`. In terms of how it reads, it feels more natural too (i.e. "these are the xpubs of this coldcard config class")